test/sampling.jl

```diff
@@ -13,6 +13,7 @@ x = [randi(10) for i = 1:n]
 @test isa(x, Vector{Int})
 @test extrema(x) == (1, 10)
 @test isapprox(proportions(x, 1:10), fill(0.1, 10), atol=5.0e-3)
+@test randi(MersenneTwister(1), 1000) == randi(MersenneTwister(1), 1000)
```

Comment:
Can you elaborate? A key (the main?) benefit of specifying the RNG is that you can get repeatable results. So this sort of check is ensuring that if you pass in the same source of randomness, you're getting back the same sample. This check did catch a bug for me - there were a few places I hadn't passed the specified RNG along to a `rand` call in a sampling method, and that caused these sorts of tests to fail.

I agree that it'd be a good idea to make sure that explicitly specifying `Base.GLOBAL_RNG` or not specifying a RNG gives the same output. Something like
```
srand(47)
x = randi(1000)
srand(47)
y = randi(Base.GLOBAL_RNG, 1000)
@test x == y
```

Which I will happily add. But that wouldn't catch the case where `randi` (or any other method) was ignoring the passed RNG and always calling `Base.GLOBAL_RNG`.

Or am I missing something obvious?